src/main/java/org/assertj/core/error/ShouldSatisfyOnlyOnce.java

```diff
@@ -0,0 +1,40 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ *
+ * Copyright 2012-2022 the original author or authors.
+ */
+package org.assertj.core.error;
+
+import org.assertj.core.util.VisibleForTesting;
+
+/**
+ * Creates an error message indicating that an assertion that verifies that requirements are not satisfied only once.
+ */
+public class ShouldSatisfyOnlyOnce extends BasicErrorMessageFactory {
+
+  @VisibleForTesting
+  public static final String REQUIREMENTS_SHOULD_BE_SATISFIED_ONLY_ONCE = "%nExpecting actual:%n  %s%nto satisfy the requirements only once but it did %s times";
```

Comment:
I think we can improve the error message:

- reword the beginning, I'm thinking of: `Expecting exactly one element of actual:%n  %s%nto satisfy the requirements but ...`
- a specific message when no elements satisfied the requirements, ex: `but no element did` or `none did`
```
Expecting exactly one element of actual:
  [e1, e2, e3, e4]
to satisfy the requirements but none did
```
- display list the offending elements when multiple elements satisfied the requirements, that will help users understanding the root cause. ex:

```
Expecting exactly one element of actual:
  [e1, e2, e3, e4]
to satisfy the requirements but these 2 elements did: 
  [e1, e3]
```